src/carousel/tp-carousel.ts

```diff
@@ -0,0 +1,413 @@
+				theArrow = arrow;
+			}
+		} );
+
+		// Return arrow.
+		return theArrow;
+	}
+
+	/**
+	 * Get current slide index.
+	 *
+	 * @return {number} Current slide index.
+	 */
+	getCurrentSlide(): number {
+		// Get current slide index.
+		return this.currentSlideIndex;
+	}
+
+	/**
+	 * Update stuff when any attribute has changed.
+	 * Example: Update sub-components.
+	 */
+	update(): void {
+		// Get sub-components.
+		const leftArrow: TPCarouselArrowElement | null = this.getArrow( 'tp-carousel-arrow[direction="previous"]' );
+		const rightArrow: TPCarouselArrowElement | null = this.getArrow( 'tp-carousel-arrow[direction="next"]' );
+
+		// Set active slide.
+		const slides: NodeListOf<TPCarouselSlideElement> | null | undefined = this.getSlideElements();
+
+		// Check if slides are available.
+		if ( slides ) {
+			slides.forEach( ( slide: TPCarouselSlideElement, index: number ): void => {
+				// Update active attribute.
+				if ( this.currentSlideIndex - 1 === index ) {
+					slide.setAttribute( 'active', 'yes' );
+				} else {
+					slide.removeAttribute( 'active' );
+				}
+			} );
+		}
+
+		// Enable / disable arrows.
+		if ( 'yes' !== this.getAttribute( 'infinite' ) ) {
+			// For the last slide.
+			if ( this.getCurrentSlide() === this.getTotalSlides() + 1 ) {
+				rightArrow?.setAttribute( 'disabled', 'yes' );
+			} else {
+				rightArrow?.removeAttribute( 'disabled' );
+			}
+
+			// For the first slide.
+			if ( 1 === this.getCurrentSlide() ) {
+				leftArrow?.setAttribute( 'disabled', 'yes' );
+			} else {
+				leftArrow?.removeAttribute( 'disabled' );
+			}
+		}
+	}
+
+	/**
+	 * Get total number of slides.
+	 *
+	 * @return {number} Total slides.
+	 */
+	getTotalSlides(): number {
+		// To get the total number of slides.
+		const slides: NodeListOf<TPCarouselSlideElement> | null | undefined = this.getSlideElements();
+
+		// Check if slides are available.
+		if ( slides ) {
+			// Tell the total number of slides.
+			return slides.length;
+		}
+
+		// Else return 0.
+		return 0;
+	}
+
+	/**
+	 * Set the current slide index.
+	 *
+	 * @param {number} index Slide index.
+	 */
+	setCurrentSlide( index: number ): void {
+		// Check if slide index is valid.
+		if ( index > this.getTotalSlides() || index <= 0 ) {
+			// Stop! It's not valid.
+			return;
+		}
+
+		// dispatch slide-set event.
+		this.dispatchEvent(
+			new CustomEvent( 'slide-set', {
+				bubbles: true,
+				detail: {
+					slideIndex: index,
+				},
+			} ),
+		);
+
+		// Set current slide index.
+		this.setAttribute( 'current-slide', index.toString() );
+	}
+
+	/**
+	 * Navigate to the next slide.
+	 */
+	next(): void {
+		// Flag that it is a programatic scroll.
+		this.flagProgramaticScroll();
+
+		// Initialize total slides variable.
+		const totalSlides: number = this.getTotalSlides();
+
+		// Check if we are at the last slide considering per view attribute.
+		if ( this.currentSlideIndex >= totalSlides - this.perView + 1 ) {
```

Comment:
@harshdeep-gill Let's hold this in a variable.
`totalSlides - this.perView + 1`